addons/sale_coupon/models/coupon_program.py

```diff
@@ -112,13 +112,16 @@ def _filter_unexpired_programs(self, order):
     def _filter_programs_on_partners(self, order):
         return self.filtered(lambda program: program._is_valid_partner(order.partner_id))
 
+    def _get_program_lines(self, order):
```

Comment:
```suggestion
    def _get_lines_suitable_for_program(self, order):
```
Or anything that could be better to be clear you don't get reward/program lines, but lines that are adapted to get a reward instead =]